velox/connectors/hive/HiveDataSink.h

```diff
@@ -238,6 +240,21 @@ class HiveInsertTableHandle : public ConnectorInsertTableHandle {
             "ensureFiles is not supported with partition keys in the data");
       }
     }
+
+    if (partitionKeyOrder_.size() > 0) {
+      // Ensure the partitionKeyOrder contains all the partition keys in
+      // inputColumns_.
+      std::string partitionKeyNames;
```

Comment:
Better to also check there is no repetition.  Something like
```c++
folly::F14FastSet<std::string> partitionKeyNames(partitionKeyOrder_.begin(), partitionKeyOrder_.end());
//...
VELOX_CHECK(partitionKeyNames.erase(inputColumn->name()) == 1);
//...
VELOX_CHECK(partitionKeyNames.empty());
```

Also in Presto we have convention to put all partitioning columns at the beginning of the `RowVector` in the right order.  Is there anything in Spark preventing us from doing so?

Reply:
@Yuhta There is no such convention in Spark to put the partition columns at the beginning of the input columns.